blockchain/blockchain.go

```diff
@@ -71,3 +71,12 @@ func (bc *BlockChain) ContainsTransaction(t *Transaction, start, stop uint32) (b
 	}
 	return false, 0, 0
 }
+
+// CopyLocalBlockByIndex returns a copy of a block in the local chain by index.
+func (bc *BlockChain) CopyBlockByIndex(i uint32) *Block {
+	blk := bc.Blocks[i]
```

Comment:
What if this is an out-of-bounds reference? I think you can also aggregate this line and the one below into one.